### PR TITLE
[vcpkg baseline][podofo] Cleanup and fixes

### PR DESCRIPTION
**Pull request by dg0yt**
Amends #31584.
Fixes baseline regression for shared triplets, cf. https://dev.azure.com/vcpkg/public/_build/results?buildId=89908.
Fixes exported config.
Makes fontconfig a default for non-windows platforms.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/podofo/portfile.cmake [hunk 1]
```diff
@@ -8,51 +8,48 @@ vcpkg_from_github(
...
+    "# Generated by CMake"
+    "include(CMakeFindDependencyMacro)
+find_dependency(Freetype)
+find_dependency(JPEG)
+find_dependency(LibXml2)
 find_dependency(OpenSSL)
-# Create imported target podofo_shared
-]]
-)
+find_dependency(PNG)
```
Comment (FrankXie05) [inline ports/podofo/portfile.cmake:43]: Optional or required? 🤔
image

https://github.com/podofo/podofo#:~:text=libjpeg%20(optional),libidn%20(optional)
https://github.com/podofo/podofo/blob/a75c4dfc69b870fab01fc9798da7619797009622/CMakeLists.txt#L172
cc @ceztko

File: ports/podofo/portfile.cmake [hunk 2]
```diff
@@ -8,51 +8,48 @@ vcpkg_from_github(
...
-        -DCMAKE_DISABLE_FIND_PACKAGE_FONTCONFIG=${PODOFO_NO_FONTMANAGER}
-        -DCMAKE_DISABLE_FIND_PACKAGE_LIBCRYPTO=${IS_WIN32}
-        -DCMAKE_DISABLE_FIND_PACKAGE_LIBIDN=ON
-        -DCMAKE_DISABLE_FIND_PACKAGE_CppUnit=ON
-        -DCMAKE_DISABLE_FIND_PACKAGE_Boost=ON
-    MAYBE_UNUSED_VARIABLES
-        CMAKE_DISABLE_FIND_PACKAGE_Boost
-        CMAKE_DISABLE_FIND_PACKAGE_CppUnit
-        CMAKE_DISABLE_FIND_PACKAGE_LIBCRYPTO
+        -DCMAKE_DISABLE_FIND_PACKAGE_Libidn=ON
```
Comment (FrankXie05) [inline ports/podofo/portfile.cmake:30]:
```suggestion
        -DCMAKE_DISABLE_FIND_PACKAGE_Libidn=ON
        -DPODOFO_BUILD_EXAMPLES=OFF
        -DPODOFO_BUILD_TEST=OFF
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/podofo/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/podofo/portfile.cmake:43]: This is the exported config for the vcpkg build. There is nothing optional. REQUIRED is implied by find_dependency. libdidn isn't used. Upstream doesn't install CMake config.

File: ports/podofo/portfile.cmake (hunk 2 above)
Comment (dg0yt) [reply ports/podofo/portfile.cmake:30]: Implied by PODOFO_BUILD_LIB_ONLY.
https://github.com/podofo/podofo/blob/a75c4dfc69b870fab01fc9798da7619797009622/CMakeLists.txt#L46C21-L49

**Review by FrankXie05 — APPROVED** [review]

File: ports/podofo/portfile.cmake (hunk 1 above)
Comment (ceztko) [reply ports/podofo/portfile.cmake:43]: These are supposed to be all optional. I'm using [`find_package`](https://cmake.org/cmake/help/latest/command/find_package.html#basic-signature) where the default should be optional unless `REQUIRED` is used.

Comment (dg0yt) [reply ports/podofo/portfile.cmake:43]: @ceztko This is understood. But it is not a topic for this PR.

**Pull request merged**